src/app/blog/[slug]/page.tsx

```diff
@@ -0,0 +1,68 @@
+import matter from "gray-matter";
+import ReactMarkdown from "react-markdown";
+import { notFound } from "next/navigation";
+import {
+  GetMarkdownFileProps,
+  MarkdownFileProps,
+  blogPostsPath,
+  generateBlogPostUrls,
+} from "./helpers";
+
+export async function generateStaticParams() {
+  const slugs = generateBlogPostUrls(blogPostsPath);
+  return slugs.map((slug: string) => {
+    return { slug };
+  });
+}
+
+export default async function BlogPost({
+  params,
+}: {
+  params: { slug: string };
+}) {
+  const { slug } = params;
+  const markdownFile = await getMarkdownFile(slug);
+
+  return (
+    <main>
+      {markdownFile?.frontmatter ? (
+        <>
+          <h1>{markdownFile.frontmatter.title}</h1>
+          <p>{markdownFile.frontmatter.date}</p>
+          <p>{markdownFile.frontmatter.tags}</p>
+          <p>{markdownFile.frontmatter.author}</p>
+          <p>Slug: {slug}</p>
+        </>
+      ) : null}
+
+      {markdownFile?.markdownBody ? (
+        <ReactMarkdown>{markdownFile.markdownBody}</ReactMarkdown>
+      ) : null}
+    </main>
+  );
+}
+
+async function getMarkdownFile(
+  slug: string
+): Promise<GetMarkdownFileProps | null> {
+  try {
+    // TODO: Rename posts folder, remove 2024
+    // Add date to MarkdownFileProps type
```

Comment:
🔧 Let's remove these comments

Reply:
More thoughts:

Changing the directory the markdown files come from isn't important because:

- eventually, we want to fetch all posts from a CMS, not from local files in the repo
- changing the file/folder structure has no effect on the paths of the website. all blog posts will still be under `/blog/fizzbuzz`